### PR TITLE
Get case names from storage API

### LINKED ISSUE #2193: Fetch casename from storage api
In the qt plotter, retrieving casenames should be backed by storage api ( ensemble names )

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #2193

### CODE REVIEW

**Review by oysteoh — APPROVED** [review]
LGTM! 🚀

**Comment by oysteoh** [comment]
Remember to squash commits and make a proper commit message! 🙏

**Comment by oysteoh** [comment]
@markusdregi @sondreso @oyvindeide This pr pulls casenames from the api instead of enkf main - and is by that the first function ported from old interface to new. If this is to be merged, we need to hide it behind a featureflag - or remove featureflag when starting ert and with that always start ert api along with ert. Anyone of you having opinions how to proceed? 

* Using featureflag 
  (+) Will enable us to partially merge code and test in production without our users noticing it
  (-) Will not enable us to remove old code as we go
  (-) Will not get much testing before the featureflags is removed
* Start ert-api when starting ert ( no featureflags )
  (+) We will have testing on an api gradually and with few interactions at first
  (+) Get rid of old code as we go and move implementation from old to new interface
  (-) Users get exposed with new functionality they do not know what is
* Featurebranch merging all of pyqt-plotter at once along with csv-export functionality
  (+) Users do not get "bothered" with unused / little used functionality
  (+) We can clean up and remove all of the old code
  (+) No featureflags
  (-) No testing before merging
  (-) Potentially a huge gap in the two codebases which is hard to merge

**Comment by frode-aarstad** [comment]
Its also worth noting that when starting the API before running ert the API locks the filesystem. So ert have to use the API to write data.

**Comment by dotfloat** [comment]
Do note that case names are just the names of directories inside the `storage/` directory. The hurdle here is that the storage directory can be set by the user with `ENSPATH`. I think using `ResConfig` and using Python to `ls` the storage should be enough.

**Comment by oysteoh** [comment]
> Its also worth noting that when starting the API before running ert the API locks the filesystem. So ert have to use the API to write data.

That is quite a huge blocker i would say? I guess this should be discussed asap how to solve as it will cause us to not use the api at all before all read / write is ported?

**Comment by oysteoh** [comment]
> Its also worth noting that when starting the API before running ert the API locks the filesystem. So ert have to use the API to write data.

Maybe a stupid question, but as long as we only read from the api - would it be possible in any way starting the api after ert and by that make it possible running both with the api read only?

File: ert_gui/tools/plot/plot_api.py [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+from typing import List
+import requests
+from ert_storage.client import Session
 
 
 class PlotApi(object):
     def __init__(self, facade):
         self._facade = facade
 
+    def _getExperiments(self) -> List[str]:
```
Comment (DanSava) [inline ert_gui/tools/plot/plot_api.py:13]: Is there a special reason this function is not called? `_get_experiments` ?

Comment (frode-aarstad) [reply ert_gui/tools/plot/plot_api.py:13]: No. I will change it

**Comment by frode-aarstad** [comment]
Jenkins test this please

**Comment by frode-aarstad** [comment]
Jenkins test this please

**Comment by frode-aarstad** [comment]
Jenkins test this please

**Comment by frode-aarstad** [comment]
Jenkins test this please

File: ert_gui/tools/plot/plot_api.py [hunk 2]
```diff
@@ -37,19 +52,20 @@ def _metadata(self, key):
             meta["data_origin"] = "Gen KW"
         return meta
 
-    def get_all_cases_not_running(self):
+    def get_all_cases_not_running(self) -> List:
```
Comment (DanSava) [inline ert_gui/tools/plot/plot_api.py:55]: Maybe we can either change the name of this function or update the API to match the previous functionality. As it is now it does not return all cases not running. Returns all cases.

**Comment by DanSava** [comment]
Jenkins test this please

**Comment by DanSava** [comment]
Jenkins test this please

**Comment by DanSava** [comment]
Jenkins test this please

File: ert_gui/tools/plot/plot_api.py [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
 class PlotApi(object):
     def __init__(self, facade):
         self._facade = facade
 
+    def _get_all_cases(self) -> List[str]:
+        cases = []
+        with Client() as client:
+            try:
+                response = client.get("/experiments")
+                expreiments = response.json()
```
Comment (oysteoh) [inline ert_gui/tools/plot/plot_api.py:22]: sorry, just a spelling error i can't let go 🙈   `experiments`

**Comment by DanSava** [comment]
Jenkins test this please

**Comment by oysteoh** [comment]
Failed with 
```
164/168 Test #151: Copy parameters from one source-fs to target-fs ......................................***Exception: SegFault  0.80 sec
Filters: Copy parameters from one source-fs to target-fs
```

**Comment by oysteoh** [comment]
should we wait with merge until we have solved the performance-issues to avoid having the pyqt plotting depending on the api?

**Comment by DanSava** [comment]
> should we wait with merge until we have solved the performance-issues to avoid having the pyqt plotting depending on the api?

Sure we can wait with the merge, but the impact of this change should not affect performance for the QT plotting significantly. Also, I think the performance issues will take some time to address.

**Comment by DanSava** [comment]
Had to revert to using `Storage.session()` in order to get a storage `Client` seems that the `Client` from `ert_storage.client`is not designed to be instantiated from ert without providing connection information, which is an argument against using.

**Comment by DanSava** [comment]
Created new PR into `api_adoption` branch (https://github.com/equinor/ert/pull/2835) closing current PR.

**Pull request closed without merging**